.github/workflows/unit-test.yml

```diff
@@ -139,4 +125,32 @@ jobs:
 
       - name: Run unit tests
         run: |
-          go test -v ./...
+          go test -v ./internal/deploy/...
+
+  querytest-test:
+    name: Run unit tests for query testing
+    needs: golangci-lint
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      checks: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          persist-credentials: false
+
+      - name: Setup Go
+        uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00 # v6.0.0
+        with:
+          go-version: "1.25.3"
```

Comment:
We probably want to test with the version in `go.mod` which currently is `1.25.0`. I'd say update there, rather than taking this (and the others in this file) back down to `1.25.0`.